armory/data/utils.py

```diff
@@ -72,3 +73,15 @@ def verify_sha256(filepath: str, hash_value: str, block_size: int = 4096):
     value = sha256(filepath, block_size=block_size)
     if value != hash_value:
         raise ValueError(f"sha256 hash of {filepath}: {value} != {hash_value}")
+
+
+def download_verify_dataset_cache(dataset_dir, checksum_file):
+    tmp_tar_fn = "tmp.tar.gz"
+    with open(checksum_file, "r") as fh:
+        url, _, hash = fh.readline().strip().split()
```

Comment:
The middle piece is file length, and would be a relatively quick check if something failed.

Reply:
Maybe create a quick "verify_size" function as well?